sinkSender.go

```diff
@@ -54,17 +48,31 @@ type Sender interface {
 	Queue(*wrp.Message)
 }
 type SinkSender struct {
-	id                               string
-	queueSize                        int
-	deliveryRetries                  int
-	maxWorkers                       int
-	disablePartnerIDs                bool
-	customPIDs                       []string
-	mutex                            sync.RWMutex
-	deliverUntil                     time.Time
-	dropUntil                        time.Time
-	deliveryInterval                 time.Duration
-	cutOffPeriod                     time.Duration
+	id                string
+	queueSize         int
+	deliveryRetries   int
+	maxWorkers        int
+	disablePartnerIDs bool
+	customPIDs        []string
+	mutex             sync.RWMutex
+	deliverUntil      time.Time
+	dropUntil         time.Time
+	deliveryInterval  time.Duration
+	cutOffPeriod      time.Duration
+	queue             atomic.Value
+	wg                sync.WaitGroup
+	workers           semaphore.Interface
+	logger            *zap.Logger
+	sink              SinkI
+	client            http.Client
+	clientMiddleware  func(Client) Client
```

Comment:
```suggestion
```

let's move this to the webhookv1 and webhookv2 sinks

kafka will use a kafka client and not a http client